### DISCUSSION TITLE
Make IColorFormat more generic

### DISCUSSION THREAD

**Discussion by Wenfengcheng** [ShapeCrawler/ShapeCrawler#496] [General]
Color can be set for many elements(like solid fill, pattern fill, picture color etc...), maybe we can not strongly bind [IColorFormat ](https://github.com/ShapeCrawler/ShapeCrawler/blob/master/src/ShapeCrawler/Drawing/IColorFormat.cs) to [Font](https://github.com/ShapeCrawler/ShapeCrawler/blob/master/src/ShapeCrawler/Texts/IFont.cs), make it more generic so it can be used in many elements.

And many variation properties(like hue, saturation, luminance, alpha, shade etc...) can be attached to color, perhaps we can also add more attributes for [IColorFormat ](https://github.com/ShapeCrawler/ShapeCrawler/blob/master/src/ShapeCrawler/Drawing/IColorFormat.cs).

```xml
<!--  SolidFill   --> 
<a:solidFill>
   <a:srgbClr val="00B050">
        <a:alpha val="51000"/>
   </a:srgbClr>
</a:solidFill>

<!--  PatternFill-->  
<a:pattFill prst="zigZag">
  <a:fgClr>
    <a:srgbClr val="333941" />
  </a:fgClr>
  <a:bgClr>
    <a:srgbClr val="3B424B" />
  </a:bgClr>
</a:pattFill>

<!--  BlipFill-->   
<a:blipFill>
  <a:blip r:embed="rId2">
    <a:duotone>
      <a:schemeClr val="accent1">
        <a:shade val="45000" />
        <a:satMod val="135000" />
      </a:schemeClr>
      <a:prstClr val="white" />
    </a:duotone>
  </a:blip>
  <a:srcRect />
  <a:stretch>
    <a:fillRect t="-9259" b="-9259" />
  </a:stretch>
</a:blipFill>
```

**Comment by AdamShakhabov** [comment]
> Color can be set for many elements(like solid fill, pattern fill, picture color etc...), maybe we can not strongly bind [IColorFormat ](https://github.com/ShapeCrawler/ShapeCrawler/blob/master/src/ShapeCrawler/Drawing/IColorFormat.cs)to [Font](https://github.com/ShapeCrawler/ShapeCrawler/blob/master/src/ShapeCrawler/Texts/IFont.cs), make it more generic so it can be used in many elements.

A good point. I agree that it should be like a general Value Object that can be used for different shape styles.

**Comment by Wenfengcheng** [comment]
[IFont](https://github.com/ShapeCrawler/ShapeCrawler/blob/master/src/ShapeCrawler/Texts/IFont.cs) aslo should be like a general Value Object.
